src/ppx/application.cpp

```diff
@@ -1257,18 +1316,22 @@ int Application::Run(int argc, char** argv)
             return EXIT_FAILURE;
         }
 
-        // Update the window size if the settings got changed due to surface requiremetns
-        {
-            int windowWidth  = 0;
-            int windowHeight = 0;
-            glfwGetWindowSize(static_cast<GLFWwindow*>(mWindow), &windowWidth, &windowHeight);
-            if ((static_cast<uint32_t>(windowWidth) != mSettings.window.width) || (static_cast<uint32_t>(windowHeight) != mSettings.window.width)) {
-                glfwSetWindowSize(
-                    static_cast<GLFWwindow*>(mWindow),
-                    static_cast<int>(mSettings.window.width),
-                    static_cast<int>(mSettings.window.height));
+#if defined(PPX_BUILD_XR)
+        if (!mSettings.enableXR) {
+            // Update the window size if the settings got changed due to surface requiremetns
+            {
+                int windowWidth  = 0;
+                int windowHeight = 0;
+                glfwGetWindowSize(static_cast<GLFWwindow*>(mWindow), &windowWidth, &windowHeight);
+                if ((static_cast<uint32_t>(windowWidth) != mSettings.window.width) || (static_cast<uint32_t>(windowHeight) != mSettings.window.width)) {
+                    glfwSetWindowSize(
```

Comment:
Can you add a `PPX_LOG_INFO` line that says the window got resized to `<width>, <height>` due to OpenXR?

Reply:
This part is skipped when XR is enabled.

Reply:
Oh, then you probably want to remove the `#if defined(PPX_BUILD_XR)`.

Reply:
fixed